src/lib/Preview.js

```diff
@@ -1421,16 +1434,16 @@ class Preview extends EventEmitter {
         this.api
             .post(`${apiHost}/2.0/events`, data, { headers })
             .then(() => {
+                this.pageTrackerReporter(true);
                 // Reset retry count after successfully logging
                 this.logRetryCount = 0;
-                this.viewer.emit('preview_event_report', true);
             })
             .catch(() => {
                 // Don't retry more than the retry limit
                 this.logRetryCount += 1;
                 if (this.logRetryCount > LOG_RETRY_COUNT) {
+                    this.pageTrackerReporter(false);
                     this.logRetryCount = 0;
-                    this.viewer.emit('preview_event_report', false);
                     return;
                 }
 
```

Comment:
Should we clear the timeout every time in the catch block. Looks like if we exit after the retry count is hit we will leave the timeout in an uncleared state?